popup/popup.js

```diff
@@ -19,69 +20,134 @@ document.addEventListener("DOMContentLoaded", function() {
-            browser.storage.local.set({ '35pokes-year': currentDate.getUTCFullYear() }, function() {});
+            browser.storage.local.set({ '35pokes-year': currentDate.getUTCFullYear() });
         }
     });
     yearElement.addEventListener('change', function() {
-        browser.storage.local.set({ '35pokes-year': this.value }, function() {});
+        browser.storage.local.set({ '35pokes-year': Number(this.value) });
+        fetchAllowedPokemonDataAndSetStorage().then(onPopupActivity);
     });
 
+    // Text Input
     const textElement = document.getElementById('text-input');
-    browser.storage.local.get(['35pokes-text'], function(items) {
+    browser.storage.local.get(['35pokes-text']).then(items => {
         if (items['35pokes-text']) {
             textElement.value = items['35pokes-text'];
         } else {
             textElement.value = '';
-            browser.storage.local.set({ '35pokes-text': '' }, function() {});
+            browser.storage.local.set({ '35pokes-text': '' });
         }
     });
     textElement.addEventListener('change', function() {
-        browser.storage.local.set({ '35pokes-text': this.value }, function() {});
+        browser.storage.local.set({ '35pokes-text': this.value });
+        fetchAllowedPokemonDataAndSetStorage().then(onPopupActivity);
     });
 
     const buttonElement = document.getElementById('code-button');
     buttonElement.addEventListener('click', function() {
-        browser.storage.local.get(['35pokes-code'], function(items) {
-            if (items['35pokes-code']) {
-                navigator.clipboard.writeText(items['35pokes-code']);
-                alert('Challenge code copied to clipboard.');
-            } else {
-                alert('No challenge code was found, please reload the page.');
-            }
+        let challengeCodePrefix = '/challenge gen9nationaldexag @@@ Z-Move Clause, -Mega, Terastal Clause, Sleep Clause Mod, Forme Clause, -Hidden Power, -Last Respects, -Kings Rock, -Shadow Tag, -Acupressure, -Battle Bond, -Quick Claw, -Razor Fang, Evasion Clause, OHKO Clause, baton pass stat trap clause, -All Pokemon, +';
+        browser.storage.local.get('35pokes-list').then(items => {
+            if(items['35pokes-list']) navigator.clipboard.writeText(challengeCodePrefix + items['35pokes-list'].join(', +')).then(() => showMessage('Copied!'));
+            else showMessage("This meta doesn't exist yet!");
         });
     });
+
+    function onPopupActivity(result) {
+        let textPrefix = result.list ? "Meta set to:<br>" : "Meta not found:<br>";
+        let textSuffix = result.text ? " " + result.text : "";
+        showMessage(textPrefix + result.month + " " + result.year + textSuffix);
+        messageTabs(result);
+    }
+
+    // This replaces alert() because alert breaks clipboard functionality on every browser.
+    // also alert in popup is unreadable for firefox users.
+    function showMessage(msg) {

```

Comment:
Partially implemented - I don't think the contrast of yellow background and bold red text is appropriate. These messages are more like logs than warnings. I'm open to adding a `level` property for the messages to specify if user attention is needed (for clipboard fails and errors, maybe for `Meta not found`).

Since this is a subjective matter, and tedious to tweak, we should merge the working implementation for now.